### DISCUSSION TITLE
Document "corresponding field" in relationship setup

### DISCUSSION THREAD

**Discussion by thepurpleblob** [directus/directus#12176] [Feature Requests]
See https://docs.directus.io/configuration/relationships/

A few lines down the docs say...

"You also have the option to create a Corresponding Field during this process. This allows you to more easily create the O2M alias (see below) that pairs with this M2O"

However, the next time it's mentioned...

"You also have the option to create a Corresponding Field during this process. This allows you to more easily create the M2O field (see above) that pairs with this O2M."

Notice see below / see above. I can't see that it's actually documented anywhere

It sounds like an interesting feature but I don't really know what it does or how to set it up.

**Comment by rijkvanzanten** [comment]
You're right! This isn't documented well in the docs. The TL;DR is that you might often want to configure the "opposite" or _corresponding_ side of a relationship (so for example when configuring an `author` m2o on `books`, you might also want to add a `written_books` o2m on `authors`). The corresponding field toggle allows you to have Directus auto-create the corresponding relationship, so you don't have to manually go in and configure the other side 🙂 

cc @erondpowell for when we're upgrading the relationship configuration docs under Configuration 👍🏻

**Reply by thepurpleblob** [reply]
Thanks - yeh, after thinking about it a bit more I eventually got the idea. I was thinking/hoping it did something else (access some other field than the id in the linked table).

**Reply by erondpowell** [reply]
Complete, exhaustive Data Model Configuration docs coming *very very* soon!!

**Comment by etmartinkazoo** [comment]
I am not seeing an option for "corresponding field" at all when I create a many to many relationship. Very frustrating:
![CleanShot 2022-07-21 at 13 15 43](https://user-images.githubusercontent.com/13721164/180274245-400904e1-33a1-4dee-ba3d-c62472446e7f.png)

**Reply by erondpowell** [reply]
hmm, interesting. I imagine this is a result of how it was configured or else the Directus version. It exists for me.
(@rijkvanzanten)

<img width="991" alt="Screen Shot 2022-07-21 at 2 02 23 PM" src="https://user-images.githubusercontent.com/62306630/180295468-fcfbbdd4-e3e1-4b9d-ba26-d9b68322d7d6.png">

One easy workaround would be to just go make an alias field in the related collection after this M2M is already setup.

**Reply by rijkvanzanten** [reply]
@erondpowell It exists when creating a new field, not when updating an existing one 👍🏻

**Reply by ffd114** [reply]
Hi @rijkvanzanten is there anyway to add corresponding field with existing field? because I did not know corresponding field exist and already input lots of data, I tried to add O2M from child collection but it stuck when trying to select parent foreign key (disabled)

**Reply by kyleva** [reply]
@ffd114 I'm late to the game but I ran into this today so figured I'd chime in for posterity. You need to create a relational field on the other model to refer back.

1. Go to the model you want a bidirectional/symmetrical relationship on 
2. Create a new **One-To-Many** field on this model in **Advanced Mode** 
3. Go to the **Relationship** configuration
4. For the the **Related Collection**, select the model you'd like to create a symmetrical relationship with
5. And for the **Foreign Key** value - select the field from the model above that refers to current model

This prevented me from losing the accumulated data and allowed me to create the bidirectional relationship retroactively.

**Reply by ffd114** [reply]
@kyleva Thanks! Will try ASAP

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
